guillotina/traversal.py

```diff
@@ -277,9 +283,12 @@ async def handler(self, request):
             resp = await apply_cors(request, resp)
 
         if not request._view_error:
-            request.execute_futures()
+            task = request.execute_futures()
         else:
-            request.execute_futures("failure")
+            task = request.execute_futures("failure")
+
+        if self.wait(request):
+            await asyncio.wait_for(task, app_settings.get("max_wait_futures"))
```

Comment:
You want the task to be cancelled if it doesn't finish in time?

if not, you should use `asyncio.wait` instead.

Reply:
yups, that is max_wait_futures settings . The idea is, if you want to wait, you want to define a max to wait. It may be conflictive if it fails and commit is already done.

Reply:
Yeah, then don't use wait_for